v1/s012-pyglet/pong.html

```diff
@@ -0,0 +1,616 @@
+                    </div>
+                    <div>
+                        Naprogramujeme si s pomocí knihovny <a href="pyglet.html">Pyglet</a>
+                        jednu z prvních videoher <a href="https://en.wikipedia.org/wiki/Pong">Pong</a>.
+                        Pong vydala společnost <a href="https://en.wikipedia.org/wiki/Atari,_Inc.">Atari</a>
+                        jako svůj první titul v roce 1972 a odstartovala tak boom herního průmyslu.
+                    </div>
+                </section>
+                <section class="col-lg-12">
+                    <h2>Konstanty a stav hry</h2>
+                    <div>
+                        Hra Pong má jednoduchá pravidla. Musíme je ale umět
+                        vyjádřit v Pythonu a to není úplně jednoduché. 
+                        Pojdmě si srovnat v hlavě, co ve hře máme.
+                        <ol>
+                            <li>Hrací pole ve tvaru obdélníku s půlící čárou.</li>
+                            <li>Míček létající určitou rychlostí po hracím poli.</li>
+                            <li>Dvě pálky pohybující se vertikálně na krajích pole.</li>
+                            <li>Dvě počítadla skóre.</li>
+                        </ol>
+                        Hra bude pro 2 hráče, nebudeme programovat chování počítače.
+                        Každý z hráčů může ovládat svou pálku stiskem kláves.
+                        Jeden hráč šipkou nahoru a šipkou dolů a druhý hráč klávesami
+                        W a S.
+
+                        Stav hry dokážeme v Pythonu vyjádřit pomocí proměnných
+                        a konstant. To dává smysl, protože některé
+                        věci se ve hře mění (poloha pálek, poloha míčku, rychlost míčku, skóre)
+                        a některé ne (velikost hrací plochy, velikost pálek a
+                        míčku, pálek a míčku, poloha a velikost počítadel skóre). 
+                        Ze složitějších datových struktur použijeme seznam
+                        (list, ten už známe) a množinu (set), která je velmi podobná
+                        matematické množině. Zjednodušeně je to seznam, který se
+                        nestará o pořadí a stejné prvky v něm mohou být právě jednou.
+                    </div>
+
+                    <div>
+                        Založte si nový skript a zkuste v Pythonu nadefinovat
+                        to jednodušší - <strong>konstanty</strong>, které 
+                        budeme v tvorbě hry potřebovat. Jen připomeňme, že
+                        názvy konstant se v Pythonu píší velkými písmeny.
+<details class="solution notes"><h4>Řešení</h4>
+<pre>
+# Velikost okna (v pixelech)
+SIRKA = 900
+VYSKA = 600
+
+VELIKOST_MICE = 20
+TLOUSTKA_PALKY = 10
+DELKA_PALKY = 100
+RYCHLOST = 200  # v pixelech za sekundu
+RYCHLOST_PALKY = RYCHLOST * 1.5  # taky v pixelech za sekundu
+
+DELKA_PULICI_CARKY = 20
+VELIKOST_FONTU = 42
+ODSAZENI_TEXTU = 30
+</pre>
+</details>
```

Comment:
Hm, tohle bych nedělal jako úlohu, spíš bych to ukázal, a na hodině prošel – společně s tím co je "pixel" a že jde rychlost měřit v pixelech za sekundu. Je prostě dost nových konceptů, na které ne každý přijde, ale s troškou vysvětlení (nebo, pro ty doma, s tímhle "řešením" a trochou času/podumání) budou jasné.